pkg/caveats/types/map.go

```diff
@@ -0,0 +1,46 @@
+package types
+
+import (
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
+)
+
+func init() {
+	subtreeFunction := cel.Function("isSubtreeOf",
+		cel.MemberOverload("subtree_maps",
+			[]*cel.Type{cel.MapType(cel.StringType, cel.AnyType), cel.MapType(cel.StringType, cel.AnyType)},
+			cel.BoolType,
+			cel.FunctionBinding(func(arg ...ref.Val) ref.Val {
+				map0 := arg[0].Value().(map[string]any)
+				map1 := arg[1].Value().(map[string]any)
+				return types.Bool(subtree(map0, map1))
+			})))
+
+	CustomTypes["__map_subtree"] = []cel.EnvOption{subtreeFunction}
```

Comment:
This is a bit of a hack... maybe we add a `types.RegisterMethodOnDefinedType` or somesuch, have it take in the function name and its member type, and produce this tree? It could even take in the args and return type.

Reply:
heh, I actually went that path originally 😅  It makes sense to me! These things would allow folks to extend SpiceDB caveat functions, same they can already add their own custom types